contractsSol5/KyberMatchingEngine.sol

```diff
@@ -172,6 +172,18 @@ contract KyberMatchingEngine is KyberHintHandler, IKyberMatchingEngine, Withdraw
         uint failedIndex; // index of error in hint
     }
 
+    /// @notice determines all the information necessary for a trade (to be returned back to network contract), or by the caller
+    /// such as what reserves were selected (their addresses and ids), what rates they offer, fee paying information
+    /// @param src Source token
+    /// @param dest Destination token
+    /// @param srcDecimals src token decimals
+    /// @param destDecimals dest token decimals
+    /// @param info array of the following: [srcAmt, networkFeeBps, platformFeeBps]
+    /// @param hint which reserves should be used for the trade
+    /// @return returns the trade wei, dest amounts, network and platform wei etc.
+    /// @dev flow is as such: src -> ETH, fee deduction (because we want to take in ETH), ETH -> dest
+    /// For ETH -> dest, if it is a split trade type, we know the reserves and whether they are fee paying, so we can do the fee deduction
+    /// However, for the other trade types, we search for the best reserve, then do the fee deduction if it is fee paying
```

Comment:
I might be a little long winded here.... =(
Maybe can keep it a bit more concise

Reply:
its good ..
will merge and cut out some words....